tooling/metamodel-generator/src/main/java/org/hibernate/processor/util/TypeUtils.java

```diff
@@ -671,6 +674,16 @@ public static boolean isMemberType(Element element) {
 		return element.getEnclosingElement() instanceof TypeElement;
 	}
 
+	public static String getGeneratedClassFullyQualifiedName(TypeElement element, String packageName, boolean jakartaDataStyle) {
+		final String packageNamePrefix = !packageName.isEmpty() ? packageName + "." : "";
+		final String className = Arrays.stream(
+						element.getQualifiedName().toString().substring( packageNamePrefix.length() ).split( "\\." ) )
+				.map( StringUtil::removeDollar )
+				.map( part -> jakartaDataStyle ? '_' + part : part + '_' )
+				.collect( Collectors.joining( "." ) );
```

Comment:
I bet if you asked IntelliJ to convert this code to a loop, it would look simpler. :)

Reply:
Yes, but I guess that `StringBuilder` will look nicer than `StringJoiner`. Maybe even slightly faster